### DISCUSSION TITLE
Streaming API causes Tesla app to report position incorrectly

### DISCUSSION THREAD

**Discussion by Sharkku** [teslamate-org/teslamate#1235] [Q&A]
When TeslaMate is connected via the streming API, the position of the car is reported incorrectly in the Tesla app. When selecting "Position" (or "Place"? - I don't know what word is used in the English version) within the app, the position is briefly shown correctly for about a second, but as soon as it updates, it goes completely wonky, showing the car in the middle of Sahara, Pacific Ocean, or something like that, moving several rhousand miles for each second, back and forth.

This does not occur if connected via the old polling method.

My car is a 2014 Model S85, with an MCU1 that was replaced about a year ago.

**Comment by adriankumpf** [comment]
This is not a bug in TeslaMate but has to be fixed by Tesla. The only option we have is to disable the usage of the Streaming API (on the TeslaMate settings page), if you want the Tesla app to show the correct location.

**Reply by Niek** [reply]
Has anyone reported this to Tesla yet? I'm surprised it's not more widely discussed, because I see this every time I check the location of the car in the app.

Maybe the streaming API is reporting lat/long offsets instead of the actual positions? Or is it completely random?

**Reply by Sharkku** [reply]
Yes, I talked to support about it. They seemed very interested until we figured out that the problem was due to me being connected via streaming API.

**Comment by scadaguru** [comment]
I have noticed same issue. Teslamate shows correct location with updated location while driving but not on phone app.

**Reply by Sharkku** [reply]
So, still an issue. I went back to the old polling API version, and haven't had an issue since.

**Comment by hefenglim** [comment]
same problem i meet, my trip data become worst, and if more error location log, it will ugly and not usable.![image](https://user-images.githubusercontent.com/21117851/132026506-03f67244-89af-4784-a2bb-714abfd60a67.jpeg)

**Reply by hefenglim** [reply]
can be ignore to log it? 
teslamate can detected abnormal gps point move unreasoable.

**Comment by ghost** [comment]
Using the streaming api is causing also other strange problems like this https://github.com/adriankumpf/teslamate/issues/247

**Comment by alexeiw123** [comment]
I don't really understand the pros/cons of streaming API vs. not, but is it possible for teslamate to not use API once the car has reported it's driving, then switch back to streaming API once it is parked, to try and achieve the best of both worlds?
